Dockerfile

```diff
@@ -0,0 +1,32 @@
+# Stage 1: Build stage
+FROM node:18-alpine as build
+
+# Set the working directory inside the container
+WORKDIR /app
+
+# Copy package.json and package-lock.json to the working directory
+COPY package.json package-lock.json ./
+
+# Install dependencies
+RUN npm install --legacy-peer-deps
+
+# Copy the entire project to the working directory
+COPY . .
+
+# Build the project
+RUN npm run build
+
+# Stage 2: Release stage
+FROM nginx:stable-alpine3.17-slim as release
+
+# Set the working directory inside the container
+WORKDIR /opt/app
+
+# Copy the NGINX configuration file
+COPY ./config/default.conf /etc/nginx/conf.d
+
+# Copy the built project from the build stage to the release stage
+COPY --from=build /app/dist/. .
```

Comment:
Where does from=build represent?

I’m confused because the built project is written to app/dist so this seems backwards

Reply:
This is a multistage build, so the container builds in stage one.  It produces the /dist folder in the 

```docker
FROM node:18-alpine as build
```

Ultimately, it builds inside the first container and then copies the file output into the nginx container (multi-stage 2).  This can ultimately be done on a build agent and inserted into a nginx container as a mount point a physical hardcopy of the nginx html directories.